### PR TITLE
Add test_pulldom.py from Cpython v3.11.2

### PR DESCRIPTION
**Pull request by Masorubka1**

### CODE REVIEW

File: Lib/test/test_pulldom.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+        evt, node = next(items)
+        self.assertEqual(pulldom.CHARACTERS, evt)
+        evt, node = next(items)
+        self.assertEqual(pulldom.END_ELEMENT, evt)
+        # XXX No END_DOCUMENT item is ever obtained:
+        #evt, node = next(items)
+        #self.assertEqual(pulldom.END_DOCUMENT, evt)
+
+    # TODO: RUSTPYTHON pulldom.parseString(SMALL_SAMPLE) return iterator with tuple with 2 elements
+    @unittest.expectedFailure
```
Comment (fanninpm) [inline Lib/test/test_pulldom.py:109]: Is `Lib/xml/dom/pulldom.py` up to date?

File: Lib/test/test_pulldom.py [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+<!-- A comment -->
+<title>Introduction to XSL</title>
+<hr/>
+<p><xdc:author xdc:attrib="prefixed attribute" attrib="other attrib">A. Namespace</xdc:author></p>
+</html>"""
+
+
+class PullDOMTestCase(unittest.TestCase):
+    # TODO: RUSTPYTHON FileNotFoundError: [Errno 2] No such file or directory (os error 2): 'xmltestdata/test.xml' -> 'None'
+    @unittest.expectedFailure
```
Comment (fanninpm) [inline Lib/test/test_pulldom.py:27]: You might want to copy this file over (along with its companion).

File: Lib/test/test_pulldom.py (hunk 1 above)
Comment (Masorubka1) [reply Lib/test/test_pulldom.py:109]: Yeap, in new version they deleted magic method __getitem__

**Review by youknowone — APPROVED** [review]
Thank you!

**Pull request merged**